src/types/sourceTargetTypes.ts

```diff
@@ -32,22 +54,34 @@ export interface ApiPromiseContextType {
   api: ApiPromise; // From @polkadot/api\
   isApiReady: boolean;
   polkadotjsUrl: string;
+  configs: Configs;
 }
 
 export enum ChainDetails {
   SOURCE = 'sourceChainDetails',
   TARGET = 'targetChainDetails'
 }
 
+// Pairs need to match with the .env value REACT_APP_PAIR
+export enum Pairs {
+  RIALTO_MILLAU = 'RIALTO_MILLAU'
+}
+
+export interface SourceState {
+  sourceConfigs: Configs;
+  sourceApiConnection: ApiPromiseConnectionType;
+  sourceChain: string;
+  sourcePolkadotjsUrl: string;
+}
+
+export interface TargetState {
+  targetConfigs: Configs;
+  targetApiConnection: ApiPromiseConnectionType;
+  targetChain: string;
+  targetPolkadotjsUrl: string;
+}
+
 export interface SourceTargetState {
-  [ChainDetails.SOURCE]: {
-    sourceApiConnection: ApiPromiseConnectionType;
-    sourceChain: string;
-    sourcePolkadotjsUrl: string;
-  };
-  [ChainDetails.TARGET]: {
-    targetApiConnection: ApiPromiseConnectionType;
-    targetChain: string;
-    targetPolkadotjsUrl: string;
-  };
+  [ChainDetails.SOURCE]: SourceState;
```

Comment:
1. We already have `Connection` which is identical.
2. We need to duplicate the type for `source` and `target`.

So essentially we now have 3 types to encode exactly the same information. What's wrong with using `Connection` here? And what's the deal with `ChainDetails` enum (i.e. what's wrong with using literal fields?)? Why this can't be simply:
```
export interface SourceTargetState {
  source: Connection,
  target: Connection,
}
```

Reply:
Tracking issue: https://github.com/paritytech/parity-bridges-ui/issues/149